File: app/src/main/java/workshop1024/com/xproject/controller/adapter/FilterListAdapter.java
package workshop1024.com.xproject.controller.adapter;

import android.databinding.DataBindingUtil;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

import workshop1024.com.xproject.R;
import workshop1024.com.xproject.controller.activity.home.NewsDetailActivity;
import workshop1024.com.xproject.databinding.FilterlistItemContentBinding;
import workshop1024.com.xproject.model.filter.Filter;
import workshop1024.com.xproject.model.news.News;

public class FilterListAdapter extends RecyclerView.Adapter<FilterListAdapter.FilterViewHolder> {
    private List<Filter> mFilterList;
    private OnFilterListDeleteListener mOnFilterListDeleteListener;

    public FilterListAdapter(List<Filter> filterList, OnFilterListDeleteListener onFilterListDeleteListener) {
        mFilterList = filterList;
        mOnFilterListDeleteListener = onFilterListDeleteListener;
    }

    @Override
    public FilterViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        FilterlistItemContentBinding filterlistItemContentBinding = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()),
                R.layout.filterlist_item_content, parent, false);
        filterlistItemContentBinding.setFilterHandlers(new FilterHandlers());
        return new FilterViewHolder(filterlistItemContentBinding);
    }

    @Override
    public void onBindViewHolder(FilterViewHolder holder, int position) {
        holder.mFilterlistItemContentBinding.setFilter(mFilterList.get(position));
    }

    @Override
    public int getItemCount() {
        return mFilterList.size();
    }

    public interface OnFilterListDeleteListener {
        void filterListItemDelete(Filter filter);
    }

    public class FilterViewHolder extends RecyclerView.ViewHolder {
        private final FilterlistItemContentBinding mFilterlistItemContentBinding;

        public FilterViewHolder(FilterlistItemContentBinding filterlistItemContentBinding) {
            super(filterlistItemContentBinding.getRoot());
            mFilterlistItemContentBinding = filterlistItemContentBinding;
        }
    }

    public class FilterHandlers {
        public void onClickDelete(View view, Filter filter) {
            if (mOnFilterListDeleteListener != null) {
                mOnFilterListDeleteListener.filterListItemDelete(filter);
            }
        }
    }
}

File: home/src/main/java/workshop1024/com/xproject/home/controller/adapter/FilterListAdapter.kt
package workshop1024.com.xproject.home.controller.adapter

import android.databinding.DataBindingUtil
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import workshop1024.com.xproject.home.R
import workshop1024.com.xproject.home.model.filter.Filter
import workshop1024.com.xproject.home.databinding.FilterlistItemContentBinding

class FilterListAdapter(private val mFilterList: List<Filter>, private val mOnFilterListDeleteListener: OnFilterListDeleteListener?) : RecyclerView.Adapter<FilterListAdapter.FilterViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FilterViewHolder {
        val filterlistItemContentBinding = DataBindingUtil.inflate<FilterlistItemContentBinding>(LayoutInflater.from(parent.context),
                R.layout.filterlist_item_content, parent, false)
        filterlistItemContentBinding.filterHandlers = FilterHandlers()
        return FilterViewHolder(filterlistItemContentBinding)
    }

    override fun onBindViewHolder(holder: FilterViewHolder, position: Int) {
        holder.mFilterlistItemContentBinding.filter = mFilterList[position]
    }

    override fun getItemCount(): Int {
        return mFilterList.size
    }

    interface OnFilterListDeleteListener {
        fun filterListItemDelete(filter: Filter)
    }

    inner class FilterViewHolder(internal val mFilterlistItemContentBinding: FilterlistItemContentBinding) : RecyclerView.ViewHolder(mFilterlistItemContentBinding.root)

    inner class FilterHandlers {
        fun onClickDelete(view: View, filter: Filter) {
            mOnFilterListDeleteListener?.filterListItemDelete(filter)
        }
    }
}
